src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs

```diff
@@ -343,7 +343,7 @@ public static void AddTopLevelSuggestionsForCursorType(IntellisenseData.Intellis
             {
                 if (info.Function.IsLambdaParam(null, argPosition) && !info.Function.HasSuggestionsForParam(argPosition) && type.IsValid)
                 {
-                    if (type.ContainsDataEntityType(DPath.Root))
+                    if (ContainsDataEntityType(type, type.MaxDepth))
```

Comment:
>ContainsDataEntityType [](http://example.com/codeflow?start=24&length=22)

This switches to a static instance call to a new method. 
But it doesn't fix the original method DType.ContainsDataEntityType... what about other callers to this method? They would still be subject to sthe stack overflow bug.